lib/puppet/provider/user/hpux.rb

```diff
@@ -63,4 +75,43 @@ def password
       return ent.passwd
     end
   end
+
+  def trusted
+      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
+      if trusted_sys.chomp == "System is not trusted."
+         "NotTrusted"
+      else
+         "Trusted"
+      end
+  end
+
+  def trust2
```

Comment:
If trust2 is purely for testing, I would move it's implementation to the spec test. Simpler interfaces are easier to maintain.

I would look at https://github.com/puppetlabs/puppet/blob/master/docs/rspec_tutorial.md for an introduction to how we write specs. There are many examples in spec/unit/* once you've read the tutorial. The provider/user/hpux_spec.rb that you modified has examples of using the #stub method, which are what you want to emulate.